extension/k8smetadata/extension.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package k8smetadata
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/aws/amazon-cloudwatch-agent/internal/k8sCommon/k8sclient"
+)
+
+const (
+	deletionDelay              = 2 * time.Minute
+	jitterKubernetesAPISeconds = 10
+)
+
+type KubernetesMetadata struct {
```

Comment:
nit: does this need to be exported? I'd imagine we can keep this lowercase

```suggestion
type kubernetesMetadata struct {
```

Reply:
Yeah, I can make this lowercase.